### PR TITLE
use images from docker_base_images repository

### PR DESCRIPTION
**Pull request by rrsettgast**
This PR deletes the docker files from this repo, and relies on images from https://github.com/GEOS-DEV/docker_base_images

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+    "build": {
+        "dockerfile": "Dockerfile",
+        "args": {
+            "docker_image_tag_hash": "b5b06ad736cc8f2df227dabf8ff1da1a2f538939"
```
Comment (TotoGaz) [inline .devcontainer/devcontainer.json:5]: An idea could be to also have a `22.04-b5b....939`?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        "vscode": {
+            "extensions": [
+                "eamodio.gitlens",
+                "fredericbonnet.cmake-test-adapter",
+                "GitHub.vscode-github-actions",
+                "GitHub.vscode-pull-request-github",
+                "ms-vscode.cpptools-extension-pack",
+                "redhat.vscode-xml"
+            ]
+        },
```
Comment (TotoGaz) [inline .devcontainer/devcontainer.json:23]: All this file is mainly if you're using `codespaces`, which is still not your case?
If you want, you can simplify it a lot, or simply embed the `b5...39` hash into your `continuousIntegration.yml` with some `echo "DOCKER_IMAGE_TAG_HASH=b5...39" >> "$GITHUB_OUTPUT"`.

**Review by TotoGaz — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (rrsettgast) [reply .devcontainer/devcontainer.json:5]: you mean to avoid the long hash?

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (rrsettgast) [reply .devcontainer/devcontainer.json:23]: I think we would want to enable codespaces even though I don't personally use it now....I would like to try it soon.

Comment (TotoGaz) [reply .devcontainer/devcontainer.json:23]: OK, it believe it's a great feature at low cost!

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (TotoGaz) [reply .devcontainer/devcontainer.json:5]: I do not know how you generate this hash. It looks to me that you use one unique image for all your builds?
In geos, we "play" with the repository name (like `ubuntu-22-gcc10...`) and the hash is unique and matches versions of the TPLs. It depends on what you want to do for Shiva. Still, it's no big deal to make this evolve in the future.

Comment (rrsettgast) [reply .devcontainer/devcontainer.json:5]: the hash is from the [docker_base_images](https://github.com/GEOS-DEV/docker_base_images) repository. I would like to standardize the images we use for testing across the project repos.

**Pull request merged**